.github/test/inspector-issues-details-uptodate-test.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const assert = require('assert').strict;
+const fs = require('fs');
+const fetch = require('node-fetch');
+
+const inspectorIssuesGathererPath = `${__dirname}/../../lighthouse-core/gather/gatherers/inspector-issues.js`;
+const inspectorIssuesGathererSource = fs.readFileSync(inspectorIssuesGathererPath, 'utf-8');
+
+describe('issueAdded types', () => {
+  /** @type {Array<LH.Crdp.Audits.InspectorIssueDetails>} */
+  let inspectorIssueDetailsTypes;
+
+  beforeAll(async () => {
+    const browserProtocolUrl =
+      'https://raw.githubusercontent.com/ChromeDevTools/devtools-protocol/master/json/browser_protocol.json';
+    const json = await fetch(browserProtocolUrl).then(r => r.json());
+
+    inspectorIssueDetailsTypes = json.domains
+      .find(d => d.domain === 'Audits')
+      .types.find(t => t.id === 'InspectorIssueDetails')
+      .properties.map(t => t.name);
+  });
+
+  it('should notify us if something changed', () => {
+    expect(inspectorIssueDetailsTypes).toMatchInlineSnapshot(`
+      Array [
+        "sameSiteCookieIssueDetails",
+        "mixedContentIssueDetails",
+        "blockedByResponseIssueDetails",
+        "heavyAdIssueDetails",
+        "contentSecurityPolicyIssueDetails",
+      ]
+    `);
+  });
+
+  it('are each handled explicitly in the gatherer', () => {
+    for (const inspectorIssueDetailsType of inspectorIssueDetailsTypes) {
+      expect(inspectorIssuesGathererSource.includes(inspectorIssueDetailsType)).toBeTruthy();
```

Comment:
I'd suggest refactoring this gatherer to use a list of supported detail types (and then create the artifacts based on that list, instead of copying the same code for each case), but that approach was decided against in code review. this is kinda awk but no more than C++ regexes so WFM...

Reply:
Considering it's the entire source file, maybe not what we want :)

Reply:
why not....

`expect(false).toBeTruthy()` doesn't seem better 😆

Reply:
Because the info you want from this test is "some value is missing". some useful info might be what the value is. but def dont want 100+ lines of code printed.

wb `if (!inspectorIssuesGathererSource.includes(inspectorIssueDetailsType)) assert.fail('missing issue type ' + inspectorIssueDetailsType)`

Reply:
> Because the info you want from this test is "some value is missing". some useful info might be what the value is.

you *also* get this information front and center with `toContain`, fwiw. for some logs that we don't really expect to fail that often ~80 lines of extra output doesn't seem that bad 🤷 though I must say I'm surprised coming from the same advocate of permanent verbose output in our smokes ;)

but ya I like your new suggestion too if your original proposal doesn't fly 👍